src/contrib/cluster/Akka.Cluster.Sharding/PersistentShardCoordinator.cs

```diff
@@ -627,22 +698,45 @@ private void HandleRegister(Register message)
                     _gracefullShutdownInProgress = _gracefullShutdownInProgress.Remove(region);
                     Update(new ShardRegionRegistered(region), e =>
                     {
-                        var isFirstRegion = _currentState.Regions.Count == 0;
                         _currentState = _currentState.Updated(e);
                         context.Watch(region);
                         region.Tell(new RegisterAck(self));
 
-                        if (isFirstRegion) AllocateShardHomes();
+                        AllocateShardHomesForRememberEntities();
                     });
                 }
             }
             else Log.Debug("ShardRegion [{0}] was not registered since the coordinator currently does not know about a node of that region", region);
         }
 
+        private void HandleGetClusterShardingStats(GetClusterShardingStats message)
+        {
+            var sender = Sender;
+            Task.WhenAll(
+                _aliveRegions.Select(regionActor => regionActor.Ask<ShardRegionStats>(GetShardRegionStats.Instance, message.Timeout).ContinueWith(r => Tuple.Create(regionActor, r.Result), TaskContinuationOptions.ExecuteSynchronously))
+                ).ContinueWith(allRegionStats =>
+                {
+                    if (allRegionStats.IsFaulted || allRegionStats.IsCanceled)
+                    {
+                        return new ClusterShardingStats(ImmutableDictionary<Address, ShardRegionStats>.Empty);
```

Comment:
For now, can you try rethrowing with a todo line and see if it fails the tests or not? If they do fail, you can let it silently swallow the exception but log an error that it shouldn't actually do that so we can track it down in the future?